intranet/apps/eighth/views/admin/blocks.py

```diff
@@ -68,6 +68,23 @@ def add_block_view(request):
 
             invalidate_model(EighthBlock)
 
+        if request.POST.get("assign_withdrawn", "off") == "on":
+            try:
+                grp = Group.objects.get(name="Withdrawn from TJ")
+                act = EighthActivity.objects.get(name="Z - Withdrawn from TJ")
+                for block in EighthBlock.objects.filter(date=fmtdate):
+                    sch_act = EighthScheduledActivity.objects.get_or_create(
+                        block=block, attendance_taken=True, administrative=True, sticky=True, activity=act, capacity=100, restricted=True
```

Comment:
I think this should be `update_or_create`, to prevent creating a duplicate `EighthScheduledActivity` should one already exist but has (a) modified attribute(s)?

```suggestion
                    sch_act = EighthScheduledActivity.objects.update_or_create(
                        block=block, activity=act,
                        defaults={"attendance_taken": True, "administrative": True, "sticky": True, "capacity": 100, "restricted": True}
```